pulp_rpm/tests/conftest.py

```diff
@@ -163,6 +162,29 @@ def _init_and_sync(
     return _init_and_sync
 
 
+@pytest.fixture
+def pulp_requests(bindings_cfg):
+    """Uses requests lib to issue an http request to pulp server using pulp_href.
+
+    Example:
+        >>> response = pulp_requests("get", "/pulp/api/v3/.../?repository_version=...")
+        >>> type(response)
+        requests.Response
+    """
+    ALLOWED_METHODS = ("get", "update", "delete", "post")
+    auth = (bindings_cfg.username, bindings_cfg.password)
+    host = bindings_cfg.host
+
+    def _pulp_requests(method: str, pulp_href: str, body=None):
+        if method not in ALLOWED_METHODS:
+            raise ValueError(f"Method should be in: {ALLOWED_METHODS}")
+        url = urljoin(host, pulp_href)
+        request_fn = getattr(requests, method)
+        return request_fn(url, auth=auth)
+
+    return _pulp_requests
+
+
```

Comment:
I think I have incidentally created a similar thing here:
https://github.com/pulp/pulp_ansible/blob/84a32718cbe18216b4067a50eea222d713c818ed/pulp_ansible/tests/functional/api/collection/v3/test_collection.py#L29

In any case, you can use `requests.request(method, *)` instead of the getattr dance.

Reply:
Nice. I guess I'll addopt the session approach, feels better that is uses the request object directly.